index.ios.js

```diff
@@ -4,6 +4,15 @@ var React = require('react-native');
 var { NativeModules, requireNativeComponent } = React;
 
 var MapMixins = {
+  addPackForRegion(mapRef, options) {
+    NativeModules.MapboxGLManager.addPackForRegion(React.findNodeHandle(this.refs[mapRef]), options);
+  },
+  getPacksWithCompletionHandler(mapRef, callback) {
```

Comment:
@bsudekum explained to me that this unfortunately still has to be asynchronous because _everything_ is asynchronous in this component, even where there’s no technical reason for asynchronicity. You may still hear me grumble loudly, but it’s a design limitation that’s apparently out of this component’s control.